app/Http/Requests/InvestmentRequest.php

```
@@ -66,6 +68,28 @@ public function rules(): array
             'currency_id' => [
                 'required',
                 Rule::exists('currencies', 'id')->where(fn ($query) => $query->where('user_id', Auth::user()->id)),
+                // Price is cast to the investment's currency (MoneyCast) while
+                // commission/tax/dividend are cast to the account's - changing the
+                // investment's currency after it's been used would silently mismatch every
+                // existing transaction's stored price against a currency it was never
+                // recorded in.
+                function (string $attribute, mixed $value, Closure $fail): void {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$attribute'. (undefined)

(UnusedFormalParameter)